airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -472,3 +528,207 @@ class Labels(GoogleAdsStream):
+            if child_slice["id_to_time"]:
+                yield child_slice
+
+    def parse_response(self, response: SearchPager, stream_slice: MutableMapping[str, Any] = None) -> Iterable[Mapping]:
+        # update records with time obtained from parent stream
+        for record in super().parse_response(response):
+            record[self.cursor_field] = stream_slice.get("id_to_time", dict()).get(record[self.primary_key[0]])
+            yield record
+
+    def _update_state(self):
+        # if parent stream was used - copy state from it, otherwise set default state
+        if self.parent_stream.state:
+            self._state = {self.parent_stream_name: self.parent_stream.state}
+        else:
+            # full refresh sync without parent stream
+            self._state = {
+                self.parent_stream_name: {
+                    self.parent_cursor_field: pendulum.today().start_of("day").format(self.parent_stream.cursor_time_format)
+                }
+            }
+
+    def _read_deleted_records(self, stream_slice: MutableMapping[str, Any] = None):
+        # yield deleted records with id and time when record was deleted
+        for deleted_record_id in stream_slice.get("deleted_ids", []):
+            yield {self.id_field: deleted_record_id, "deleted_at": stream_slice["id_to_time"].get(deleted_record_id)}
+
+    def read_records(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_slice: MutableMapping[str, Any] = None, **kwargs
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        This method is overridden to read records using parent stream
+        """
+        self.incremental_sieve_logger.bump()
+        self.incremental_sieve_logger.info(f"Started reading records for slice: {stream_slice}")
+
+        # if state is present read records by ids from slice otherwise full refresh sync
+        yield from super().read_records(sync_mode, stream_slice=stream_slice)
+
+        # yield deleted items
+        yield from self._read_deleted_records(stream_slice)
+
+        self._update_state()
+
+    def get_query(self, stream_slice: Mapping[str, Any] = None) -> str:
+        query = GoogleAds.convert_schema_into_query(
+            schema=self.get_json_schema(), report_name=self.name, id_field=self.id_field, id_list=stream_slice.get("updated_ids")
+        )
+        return query
+
+
+class AdGroupCriterions(IncrementalEventsStream):
+    """
+    Ad Group Criterions stream: https://developers.google.com/google-ads/api/fields/v14/ad_group_criterion
+    """
+
+    transformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization)
+    primary_key = ["ad_group_criterion.resource_name"]
+    parent_id_field = "change_status.ad_group_criterion"
+    id_field = "ad_group_criterion.resource_name"
+    resource_type = "AD_GROUP_CRITERION"
+    cursor_field = "change_status.last_change_date_time"
+
+
+class AdListingGroupCriterions(AdGroupCriterions):
```

Comment:
what is the difference between this class and `AdGroupCriterion`? the docstring is the same

Reply:
These streams have different configurations of schemas, AdListingGroupCriterion request fields about listing_group.

Reply:
ok, can you list that in the docstring with the appropraite URL?

Reply:
Added